### PR TITLE
Added authentication and toasts

### PR DESCRIPTION
**Pull request by nikhiljugale007**
Added authentication , added toast while adding and removing items to cart and wishlist.

##  **In the preview if you are not able to login using adarshbalika credentials, singup with your email and then login using that credentials. Do not refresh page in between signup and login.**

Files to check:
Login.jsx
SignUp.jsx
Profile.jsx
apicalls.js
ProductCard.jsx
CartProductCard.jsx

### CODE REVIEW

File: src/components/form_input/FormInput.jsx [hunk 1]
```diff
@@ -0,0 +1,23 @@
+import "./FormInput.css";
+const FormInput = ({ label, value, onChangeFunction, type, required }) => {
+	return (
+		<div>
+			<div className="input-container">
+				<label className="label text-bold" htmlFor={label}>
```
Comment (Naman-Saxena1) [inline src/components/form_input/FormInput.jsx:6]: If you are trying to dynamically give className, then I think it should be like this -
className=\` ${label} text-bold\`.
And I am unable to find class label in any CSS files

File: src/components/cart_product_card/CartProductCard.jsx [hunk 2]
```diff
@@ -5,20 +5,42 @@ import {
 	changeQuantity,
 } from "../../api/apicall";
 import { useProductContext } from "../../context/ProductContext";
+import { Toast } from "../toast/Toast";
```
Comment (Naman-Saxena1) [inline src/components/cart_product_card/CartProductCard.jsx]: Suggestion, you can export components and context files from index.js and throughout your codebase only import from index.js

**Review by Naman-Saxena1 — CHANGES_REQUESTED** [review]
In preview, I am not able to click on Create new account button on /login page, you can cross check that.

File: src/api/apicall.js [hunk 3]
```diff
@@ -3,6 +3,24 @@ const auth =
 	"eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJfaWQiOiI4MTAyMWNjNC00YjFkLTQyOGItYjJmMC0wNjhkYTQ4YTk4MzQiLCJlbWFpbCI6ImFkYXJzaGJhbGlrYUBnbWFpbC5jb20ifQ.45ynQ6aZhoM1zsNwIKCYR_IATaszKn0ssvnPPQkKL8E";
 const headers = { authorization: auth };
 
+const signupuser = async (user) => {
+	try {
+		const response = await axios.post("/api/auth/signup", { user });
+		return { token: response.data.encodedToken, success: true };
+	} catch (err) {
+		console.log(err);
```
Comment (varun1233) [inline src/api/apicall.js]: Remove console

File: src/components/cart_product_card/CartProductCard.jsx [hunk 4]
```diff
@@ -33,55 +55,60 @@ const CartProductCard = ({ product }) => {
...
-							Move to wishlist
+					<div className="card-footer-btn">
+						{checkItemInWishlist() ? (
+							<button className="btn btn-primary">Already Wishlisted</button>
+						) : (
+							<button className="btn btn-primary" onClick={addItemToWishlist}>
+								Move to wishlist
+							</button>
+						)}
+						<button className="btn btn-outlined" onClick={removeItemToCart}>
```
Comment (sri-rishi) [inline src/components/cart_product_card/CartProductCard.jsx]: Suggestion: `removeItemToCart` can be named as `removeItemFromCart`  or something better

File: src/components/cart_product_card/CartProductCard.jsx [hunk 5]
```diff
@@ -33,55 +55,60 @@ const CartProductCard = ({ product }) => {
...
+							Remove
```
Comment (sri-rishi) [inline src/components/cart_product_card/CartProductCard.jsx]: Suggestion: same here name button something like `Remove from Wishlist`

**Review by sri-rishi — APPROVED** [review]
Great work, just some naming suggestion which I think can be done otherwise code looks good to merge

**Review by yaswanthmyneni — APPROVED** [review]
The code is clean to me, validating user credentials before signing in is what I haven't done in my project, now I will do it, you can merge it.

File: src/api/apicall.js [hunk 6]
```diff
@@ -37,6 +54,7 @@ const addToWishlist = async (product) => {
 		return { wishlist: [], success: false };
 	}
 };
+
 const removeFromWishlist = async (item_id) => {
```
Comment (varun1233) [inline src/api/apicall.js]: Try to use same convention for naming variables, in other places you are using camelCase

**Review by varun1233 — CHANGES_REQUESTED** [review]
Code looks good. Suggested some chanes

File: src/components/form_input/FormInput.jsx (hunk 1 above)
Comment (nikhiljugale007) [reply src/components/form_input/FormInput.jsx:6]: htmlfor is for connecting the label with the input.

File: src/components/cart_product_card/CartProductCard.jsx (hunk 2 above)
Comment (nikhiljugale007) [reply src/components/cart_product_card/CartProductCard.jsx]: Implemented suggestion in new commit.

File: src/components/cart_product_card/CartProductCard.jsx (hunk 4 above)
Comment (nikhiljugale007) [reply src/components/cart_product_card/CartProductCard.jsx]: Implemented suggestion in new commit.

File: src/components/cart_product_card/CartProductCard.jsx (hunk 5 above)
Comment (nikhiljugale007) [reply src/components/cart_product_card/CartProductCard.jsx]: implemented suggestion in new commit.

File: src/api/apicall.js (hunk 3 above)
Comment (nikhiljugale007) [reply src/api/apicall.js]: removed console in new commit.

File: src/api/apicall.js (hunk 6 above)
Comment (nikhiljugale007) [reply src/api/apicall.js]: I will keep this in mind. I am trying to follow camelCase everywhere.

**Pull request merged**